### PR TITLE
Remove log4j

### PR DESCRIPTION
**Pull request by Duhemm**

### CODE REVIEW

File: backend/src/main/scala/bloop/logging/BloopLogger.scala [hunk 1]
```diff
@@ -1,89 +1,67 @@
...
-    val appender = OutputStreamAppender
-      .newBuilder()
-      .setName(s"$name-$nameSuffix")
-      .setFilter(filter)
-      .setLayout(layout)
-      .setTarget(stream)
-      .build()
-
-    appender
+  private def printTrace(line: String): Unit = {
```
Comment (jvican) [inline backend/src/main/scala/bloop/logging/BloopLogger.scala]: Shall we add a method for `Throwable` here?

Comment (jvican) [reply backend/src/main/scala/bloop/logging/BloopLogger.scala]: Oh no, just realized that the method targeting `Throwable` comes inherited.

File: backend/src/main/scala/bloop/logging/BloopLogger.scala [hunk 2]
```diff
@@ -1,89 +1,71 @@
...
+      print(prefix + exception.toString(), printTrace)
+      exception.getStackTrace.foreach(ste => print("\t" + ste.toString, printTrace))
+      trace("Caused by: ", exception.getCause)
+    }
+  }
 
-    outAppender.start()
-    errAppender.start()
+  private def print(msg: String, fn: String => Unit): Unit = {
+    if (msg == null) ()
```
Comment (jvican) [inline backend/src/main/scala/bloop/logging/BloopLogger.scala]: This implementation is pretty weird. why `.lines` and why the NPE guard? Is there a way we can remove that from this core logging API endpoint?

Comment (Duhemm) [reply backend/src/main/scala/bloop/logging/BloopLogger.scala]: `.lines` is to have the prefix on every line (think `log.error("foo\nbar")`). The NPE guard is to make sure `.lines` won't crash. There are cases where `exception.getMessage` is `null`.

File: frontend/src/test/scala/bloop/nailgun/BasicNailgunSpec.scala [hunk 3]
```diff
@@ -119,7 +123,54 @@ class BasicNailgunSpec extends NailgunTest {
...
+
+      assertTrue("`logger1` received messages of `logger2`",
+                 !msgs1.exists(_.startsWith("Projects loaded from")))
+      assertEquals("`logger` didn't receive verbose messages",
+                   3,
+                   msgs1.count(_.startsWith(s"${debugPrefix}Elapsed:")).toLong)
+      assertTrue("`logger2` received messages of `logger1`",
+                 !msgs2.exists(_.startsWith("Compiling")))
+      assertTrue("`logger2` received verbose messages of `logger1`",
+                 !msgs2.exists(_.startsWith(debugPrefix)))
```
Comment (jvican) [inline frontend/src/test/scala/bloop/nailgun/BasicNailgunSpec.scala]: :tada:

**Pull request merged**